thewalrus/decompositions.py

```diff
@@ -202,6 +202,15 @@ def takagi(A, svd_order=True):
         vals, U = takagi(Amr, svd_order=svd_order)
         return vals, U * np.exp(1j * phi / 2)
 
+    # If the matrix is diagonal, Takagi decomposition is easy
+    if np.allclose(A, np.diag(np.diag(A))):
+        d = np.diag(A)
+        U = np.diag(np.exp(1j * 0.5 * np.angle(d)))
+        l = np.abs(d)
+        if svd_order is False:
```

Comment:
You need to correct this. First you need to sort `l` and rearrange `U` accordingly. Then you can decide if you flip the order. As is, this will not return the takagi singular values sorted correctly.